smac/facade/smac_facade.py

```diff
@@ -207,75 +253,100 @@ def __init__(self,
-                                              seed=rng.randint(MAXINT),
-                                              pca_components=scenario.PCA_DIM,
-                                              log_y=scenario.transform_y in ["LOG", "LOGS"],
-                                              num_trees=scenario.rf_num_trees,
-                                              do_bootstrapping=scenario.rf_do_bootstrapping,
-                                              ratio_features=scenario.rf_ratio_features,
-                                              min_samples_split=scenario.rf_min_samples_split,
-                                              min_samples_leaf=scenario.rf_min_samples_leaf,
-                                              max_depth=scenario.rf_max_depth)
+            for key, value in {
+                'log_y': scenario.transform_y in ["LOG", "LOGS"],
+                'num_trees': scenario.rf_num_trees,
+                'do_bootstrapping': scenario.rf_do_bootstrapping,
+                'ratio_features': scenario.rf_ratio_features,
+                'min_samples_split': scenario.rf_min_samples_split,
+                'min_samples_leaf': scenario.rf_min_samples_leaf,
+                'max_depth': scenario.rf_max_depth,
+            }.items():
+                if key not in model_def_kwargs:
+                    model_def_kwargs[key] = value
+            model = RandomForestWithInstances(**model_def_kwargs)
+        elif inspect.isclass(model):
+            model = model(**model_def_kwargs)
+        else:
+            raise TypeError(type(model))
+
         # initial acquisition function
+        acq_def_kwargs = {'model': model}
+        if acquisition_function_kwargs is not None:
+            acq_def_kwargs.update(acquisition_function_kwargs)
         if acquisition_function is None:
             if scenario.transform_y in ["LOG", "LOGS"]:
-                acquisition_function = LogEI(model=model)
+                acquisition_function = LogEI(**acq_def_kwargs)
             else:
-                acquisition_function = EI(model=model)
-
-        # inject model if necessary
-        if acquisition_function.model is None:
-            acquisition_function.model = model
+                acquisition_function = EI(**acq_def_kwargs)
+        elif inspect.isclass(acquisition_function):
+            acquisition_function = acquisition_function(**acq_def_kwargs)
+        else:
+            raise TypeError(
+                "Argument acquisition_function must be None or an object implementing the "
+                "AbstractAcquisitionFunction, not %s."
+                % type(acquisition_function)
+            )
 
         # initialize optimizer on acquisition function
+        acq_func_opt_kwargs = {
+            'acquisition_function': acquisition_function,
+            'config_space': scenario.cs,
+            'rng': np.random.RandomState(seed=rng.randint(MAXINT)),
```

Comment:
Why is it necessary to generate a new rs here? In line 233 we simply use directly the rng

Reply:
It's not. I just changed that.